src/lib/popper.ts

```diff
@@ -92,6 +96,30 @@ export class PopperManager {
       container.classList.add('popper-open')
     }
   }
+  // func to detreimne if the popper will overflow the body boundary
+  private fixPopperPosition(data: Popper.Data, popperElement: HTMLDivElement) {
+    const toolsPanel = popperElement.querySelector(
+      '.tools-panel'
+    ) as HTMLElement
+    const totalWidth = data.offsets.popper.left + toolsPanel.clientWidth
+    const arrow = popperElement.querySelector('.popper-arrow') as HTMLElement
+    // the following will always change the placement to bottom,
+    // for better positioning of the arrow and the popper
+    if (data.originalPlacement != 'bottom') {
+      data.instance.options.placement = 'bottom'
+      data.instance.update()
+    }
+    if (totalWidth > window.innerWidth) {
+      // the 40 is the size of the resizer button
+      popperElement.style.left =
+        (window.innerWidth - totalWidth - 40).toString() + 'px'
+      if (arrow) {
+        // the 30 is the size of the resizer button - the arrow margin value
+        arrow.style.left =
+          Math.abs(window.innerWidth - totalWidth - 30).toString() + 'px'
+      }
+    }
+  }
```

Comment:
we can let the library do that for us instead depending on the resizer button. so to fix the overflow issue use position relative to both: [citation-editor](https://github.com/Atypon-OpenSource/manuscripts-body-editor/blob/e275986540025765a3b45e7f87b15bdf0d311ef7/styles/popper.css#L198) & [link-editor](https://github.com/Atypon-OpenSource/manuscripts-body-editor/blob/e275986540025765a3b45e7f87b15bdf0d311ef7/styles/popper.css#L190C9-L190C21) and set placement to auto

Reply:
this would be a much better solution